azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CAutoConfiguration.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.b2c;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import static com.microsoft.azure.spring.autoconfigure.b2c.AADB2CProperties.POLICY_SIGN_UP_OR_SIGN_IN_NAME;
+import static com.microsoft.azure.spring.autoconfigure.b2c.AADB2CProperties.POLICY_SIGN_UP_OR_SIGN_IN_REDIRECT_URL;
+
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnProperty(prefix = AADB2CProperties.PREFIX, value = {
+        "tenant",
+        "client-id",
+        POLICY_SIGN_UP_OR_SIGN_IN_NAME,
+        POLICY_SIGN_UP_OR_SIGN_IN_REDIRECT_URL,
+})
+@EnableConfigurationProperties(AADB2CProperties.class)
+public class AADB2CAutoConfiguration {
```

Comment:
In Java naming convention, this might be `AadB2c`.

Reply:
Does this config need some condition on class?

Reply:
For noun, Yes. But here is abbreviation, like `java.net.URI`.